web/src/browser_history.js

```diff
@@ -73,7 +74,7 @@ export function update(new_hash) {
 export function exit_overlay() {
     if (hash_util.is_overlay_hash(window.location.hash) && !state.changing_hash) {
         ui_util.blur_active_element();
-        const new_hash = state.hash_before_overlay || "#";
+        const new_hash = state.hash_before_overlay || `#${user_settings.default_view}`;
```

Comment:
Why is this the default view hash rather than the empty string?

Reply:
Since this function uses `update` which uses `window.location.hash`.